docs/source/install.rst

```diff
@@ -294,40 +274,167 @@ should run::
    whatever packages you just installed.
 
```

Comment:
```suggestion

.. note::
   The command given to clone the PETSc repository in the :ref:`Installing PETSc<install_petsc>` section include the `--depth 1` argument, which forces git to only clone the most recent commit. This massively reduces the amount of data that must be downloaded, but also means that the entire git history is missing. If you are developing PETSc then you may want to use a larger depth in this step, or remove this argument entirely.
```

Reply:
I'll mention something like this. I want to figure out the right git magic to undo `--depth 1`.

Reply:
Actually I will add this in https://github.com/firedrakeproject/firedrake/pull/4193 because using a custom PETSc is quite complicated.